File: 프로그래머스/0/120840. 구슬을 나누는 경우의 수/구슬을 나누는 경우의 수.cs
using System;

public class Solution 
{
    public double solution(int balls, int share) 
    {
        
        double n = 1;
        double m = 1;
        double r = 1;
        
        
        for(double i=1; i<=balls; i++)
        {
            n *= i;
        }
        for(double j=1; j<=share; j++)
        {
            m *= j;
        }
        if(balls > share)
        {
            for(double k=1; k<=(balls - share); k++)
            {
                r *= k;
            }     
        }
        else if(balls == share)
        {
            r = 1;
        }
        
        double combination =  Math.Round(n / (m * r));
        return combination;
    }
}

File: 프로그래머스/0/120840. 구슬을 나누는 경우의 수/구슬을 나누는 경우의 수.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

int solution(int balls, int share) 
{
    long double n = 1;
    for(int i=share + 1; i<=balls; i++)
    {
        n *= i;
    }
    for(int i=1; i<=balls - share; i++)
    {
        n /= i;
    }
    cout << n << endl;
    return n;
}